File: src/abstracts/command.py
from abc import ABC, abstractmethod

class Command(ABC):
  #Constructor
  def __init__(self, name, description, triggers):
    #Command dictionary
    self.name = name
    self.description = description
    self.triggers = triggers
    
    super().__init__()

  #Getters
  def getName(self):
    return self.name

  def getDescription(self):
    return self.description

  def getTriggers(self):
    return self.triggers
  
  @abstractmethod
  async def execute(self, message, client, args):
    pass

File: src/abstracts/Command.ts
// Abstract class to set up commands

import { Message } from 'discord.js';

export abstract class Command {
  private name: string;
  private description: string;
  private triggers: string[];

  constructor(name: string, description: string, triggers: string[]) {
    this.name = name;
    this.description = description;
    this.triggers = triggers;
  }

  //Get the Name of the command
  getName(): string {
    return this.name;
  }

  //Get the Description of the command
  getDescription(): string {
    return this.description;
  }

  //Get the Triggers for the command
  getTriggers(): string[] {
    return this.triggers;
  }

  //Abstract execute function for executing command
  abstract execute(message: Message, args: string[]): Promise<void>;
}
